### PR TITLE
Cleaning 2x Routes

### LINKED ISSUE #495: Clean up 2.x routes
The `client/app.js` file currently contains many routes directly ported from 1.x that no longer point to updated modules. 

In order to make sure that all 2.x code and modules are correctly integrated we should remove all outdated references - these can be added back in as they are developed/ upgraded to 2.x. 

This refactor has two steps: 
- Remove all references to 1.x modules that are not yet upgraded to 2.x
- Remove all '/routes/ in state name - these should be completely seperate from the `url`
  (@todo add example of this)

This base `app.js` will allow us to split up application routing into individual modules creating a much more modular routing structure that should not conflict as much during development. (Proposed https://github.com/IMA-WorldHealth/bhima-2.X/issues/494)

### PR DESCRIPTION
**Pull request by lomamech**
Cleaning 2x Routes

Closes #495.

@jniles

### CODE REVIEW

**Comment by jniles** [comment]
@lomamech, very good work!  If all tests pass, this will be merged in.

Edit: As discussed, it would be good to change slashes in the state names to `camelCase`.  For example: `/inventory/configuration` --> `inventoryConfiguration`.

**Comment by lomamech** [comment]
Improve param state in app.js

@jniles

File: client/src/js/app.js [hunk 1]
```diff
@@ -143,66 +102,35 @@ function bhimaConfig($stateProvider, $urlRouterProvider, $urlMatcherFactoryProvi
...
   /* location routes */
 
-  .state('/locations', {
+  .state('locations', {
     url : '/locations',
     controller : 'LocationController as LocationCtrl',
     templateUrl: 'partials/locations/locations.html'
   })
-  .state('locations/village', {
+  .state('locationsVillage', {
```
Comment (jniles) [inline client/src/js/app.js:118]: Well done :+1:

File: client/src/js/app.js [hunk 2]
```diff
@@ -757,201 +440,6 @@ function bhimaConfig($stateProvider, $urlRouterProvider, $urlMatcherFactoryProvi
...
-  })
-  .state('/reports/patient_visit_status', {
-    controller : 'ReportPatientVisitStatus',
-    templateUrl : '/partials/reports/patient_visit_status/patient_visit_status.html'
-  })
-  .state('/reports/stock_entry', {
-    controller : 'ReportStockEntryController',
-    templateUrl : 'partials/reports/stock/stock_entry/stock_entry.html'
-  })
   .state('/error403', {
```
Comment (jniles) [inline client/src/js/app.js]: Whoops, looks like you forgot this `/`.  Could you remove it?  There should not be any slashes in the state names.

**Comment by jniles** [comment]
LGTM :+1:.  Once all tests pass, this can be merged!

**Pull request merged**